src/app/components/upload-event/upload-event.component.html

```diff
@@ -1,50 +1,75 @@
 <div class="container">
-  <h2>Upload an Event</h2>
-  <form (ngSubmit)="onSubmit()" #eventForm="ngForm">
-    <div class="form-group">
-      <label for="userId">User ID</label>
-      <input #userIdField type="text" class="form-control" id="userId" required name="userId" [(ngModel)]="eventModel.userId" #userId="ngModel">
-      <div [hidden]="userId.valid || userId.pristine" class="alert alert-danger">
-        User ID is required
-      </div>
-    </div>
-
-    <div class="form-group">
-      <label for="userName">User Name</label>
-      <input type="text" class="form-control" id="userName" required name="userName" [(ngModel)]="eventModel.userName" #userName="ngModel">
-      <div [hidden]="userName.valid || userName.pristine" class="alert alert-danger">
-        User Name is required
-      </div>
-    </div>
-
-    <div class="form-group">
-      <label for="group">Group</label>
-      <input type="text" class="form-control" id="group" required name="group" [(ngModel)]="eventModel.group" #group="ngModel">
-      <div [hidden]="group.valid || group.pristine" class="alert alert-danger">
-        Group is required
-      </div>
-    </div>
-
-    <div class="form-group">
-      <label for="assetId">Asset ID</label>
-      <input type="number" class="form-control" id="assetId" required name="assetId" min="0" [(ngModel)]="eventModel.assetId" #assetId="ngModel">
-      <div [hidden]="assetId.valid || assetId.pristine" class="alert alert-danger">
-        Asset ID is required
-      </div>
-    </div>
-
-    <div class="form-group">
-      <label for="assetTitle">Asset Title</label>
-      <input type="text" class="form-control" id="assetTitle" required name="assetTitle" [(ngModel)]="eventModel.assetTitle" #assetTitle="ngModel">
-      <div [hidden]="assetTitle.valid || assetTitle.pristine" class="alert alert-danger">
-        Asset Title is required
-      </div>
-    </div>
-
-    <button type="submit" class="btn btn-success" [disabled]="!eventForm.form.valid">
-      Generate
-    </button>
-
-  </form>
+    <h2>Upload an Event</h2>
+
+    <form [formGroup]="eventForm" (ngSubmit)="onSubmit()">
+        <div class="form-group">
+            <label class="center-block">User ID:
+                <input class="form-control" formControlName="userId" required #userIdInput>
+            </label>
+        </div>
+        <div>
+            <p *ngIf="!(eventForm.get('userId').valid || eventForm.get('userId').pristine)" class="alert alert-danger">User ID is required</p>
```

Comment:
This logic is repeated for each field so you can probably extract into a method

e.g.

```
shouldShowFieldValidationMethod(fieldname: string)
```

Reply:
Although before you do this ensure you have tests checking that each of validation errors show when they should so you know the refactoring has broken anything.